### ISSUE TITLE
Questions regrading text chunking

### ISSUE THREAD

**Issue by Jack-ZC8** [Teddy-XiongGZ/MedRAG#9]
I wanted to take a moment to express my sincere appreciation for the outstanding work. Your code is truly impressive. The overall structure is well-thought-out, and the implementation is clean and efficient!

I have a couple of questions:

1. **Chunk Size and Overlap**: What was the reasoning behind the `chunk_size=1000` and `chunk_overlap=200` parameters? I'd be interested to learn about your experience with these settings and any trade-offs you considered.

2. **PubMed Dataset Chunking**: Given the potentially large size of the PubMed dataset, do you think it would be worth performing chunking it?

**Comment by gzxiong** [comment]
Thanks for your interest in our work. Regarding your questions:

1. Chunk size and overlap: we used chunk_size=1000 because that roughly ensures each chunk has less than 512 tokens (considering the fact that each token contains several characters and the BERT-based retrievers have a context window of 512 tokens). chunk_overlap=200 was chosen following the default setting in langchain.

2. PubMed Dataset Chunking: We did not consider chunking the PubMed dataset because each snippet in it is a semantically complete abstract. In addition, its average length is 296 tokens, which is compatible with the context window size of retrievers. Chunking may help with very long abstracts in PubMed, but they represent a small fraction of the abstracts and we do not anticipate a significant performance gain from this.
